### ISSUE TITLE
Handle Text Alignment when parsing HTML

### ISSUE THREAD

**Issue by leo-paz** [TypeCellOS/BlockNote#636]
**Is your feature request related to a problem? Please describe.**
When copying over work from a .docx document in a word editor text alignment is never respected. Everything becomes left aligned. I'm assuming this would involve changes to how HTML is parsed.

**Describe the solution you'd like**
Would be nice for it to respect the text alignment given to it in the initial editor

**Additional context**
![image](https://github.com/TypeCellOS/BlockNote/assets/26093865/daf8cfb8-d038-4934-b7dc-946348e7e18d)
![image](https://github.com/TypeCellOS/BlockNote/assets/26093865/2e07d721-c197-4bac-bc29-d62720f67dc1)

**Comment by linxianxi** [comment]
same here

**Comment by matthewlipski** [comment]
This is because unfortunately there isn't a lot of consistency with how text styles are handled when copying to the clipboard for different editors. Things like bold and italic are no problem since they have corresponding HTML tags, but text color and alignment could be marked using e.g. `data-` attributes or inline styles. So basically we would have to make specific cases for various popular editors, which we haven't gotten around to yet.

**Comment by cmawhorter** [comment]
is there a reason tiptap's [generateHTML](https://tiptap.dev/docs/editor/api/utilities/html) isn't used? from a quick look at the blocknote code, it seems like blocks are tiptap nodes, which are just prosemirror nodes. 

i've been evaluating replacements for a very old custom tinymce editor and i was kinda surprised by the lack of support for html. but after finding this issue and re-reading the docs, it seems i misunderstood the goal of blocknote which is to be a literal rich-text editor, and not html content authoring/editing. 

that also explains why the docs recommend persisting document json, which i was initially confused by.

that's a shame because blocknote is essentially drop-in if you already use one of the supported ui component libs. 

this is my ignorance showing, but surely tiptap or pm has already solved this problem of transparently converting between html and blocks/nodes?

**Comment by matthewlipski** [comment]
You're right with a lot of these points and yes, BlockNote is generally intended for documents that stay editable & interactive rather than outputting/exporting static content. The main concern for us with exporting fully styled HTML was whether we want to modify it vs what you see in the DOM, as that has a lot of extra stuff that is no longer needed once the document is static (things like block ID attributes, React wrapper `div`s, etc).

However, with 0.15.0 you're now able to effectively export BlockNote documents to HTML without losing styling and nesting information:) We ended up just exporting basically 1:1 HTML to what's in the DOM so it's a bit messy, but it should fit your use case now.

**Comment by matthewlipski** [comment]
So converting between HTML and JSON is indeed handled by TipTap/ProseMirror, the issue is that you still need to give it instructions for how to parse the HTML. When you're working within the same schema, this is pretty cut and dry as you know how each node is represented in HTML as well as JSON. However, when inserting external HTML into your document, from Word of other editors/websites, you don't necessarily know how that information is represented.

For example, Google Docs shoves a whole bunch on inline styles into their HTML to make it match the actual document as closely as possible, whereas Notion doesn't have any information regarding things like text/bg colors in their HTML. So you kind of have to tell TipTap/PM how to parse certain nodes (or marks in this example) by accounting for these cases individually.

**Comment by YousefED** [comment]
@cmawhorter you might be interested in the recently release `server-util` package for this as well: https://www.blocknotejs.org/docs/editor-api/server-processing

**Comment by cmawhorter** [comment]
you guys are on the ball. nice! i'm going to take a look at the server-util package and it looks promising. thanks for your work on it.
